### PR TITLE
[ACR] Add acr login fallback on wincred errors, add registry request logs

### PR DESCRIPTION
**Pull request by djyou**
Fixes https://github.com/Azure/azure-cli/issues/4843

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -251,14 +251,35 @@ def acr_login(registry_name, resource_group_name=None, username=None, password=N
...
         p = Popen(["docker", "login",
                    "--username", username,
                    "--password", password,
-                   login_server])
-        p.wait()
+                   login_server], stderr=PIPE)
+        _, stderr = p.communicate()
+
+    if stderr:
+        if b'error storing credentials' in stderr and b'stub received bad data' in stderr \
```
Comment (sajayantony) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: Is docker localized or can we take dependency on this?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py [hunk 2]
```diff
@@ -271,3 +292,45 @@ def acr_show_usage(registry_name, resource_group_name=None):
...
     return client.list_usages(resource_group_name, registry_name)
+
+
+def _check_wincred(login_server):
+    import platform
+    if platform.system() == 'Windows':
+        import json
+        from os.path import expanduser, isfile
+        home = expanduser("~").strip('\\')
+        config_path = '{}{}'.format(home, '\\.docker\\config.json')
```
Comment (sajayantony) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: path.join?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py (hunk 1 above)
Comment (djyou) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: Docker CLI doesn't have localization at this point so this should be good.

**Comment by derekbekoe** [comment]
Please sync with dev branch due to the merge of https://github.com/Azure/azure-cli/pull/4904 and resolve conflicts.

**Comment by tjprescott** [comment]
Due to recent changes to the CLI core, you will need to rebase your changes against the latest in the dev branch.

**Pull request closed without merging**